ambry-protocol/src/main/java/com.github.ambry.protocol/AdminRequest.java

```diff
@@ -81,6 +77,7 @@ public long writeTo(WritableByteChannel channel) throws IOException {
     long written = 0;
     if (bufferToSend == null) {
       serializeIntoBuffer();
+      bufferToSend.flip();
```

Comment:
Ok, i saw your new version removed that line. So why you want to do flipping here?